pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -106,10 +104,107 @@ func (d *DefaultEvictor) addAllConstraints(logger klog.Logger, handle frameworkt
 		d.constraints = append(d.constraints, constraints...)
 	}
 	d.constraints = append(d.constraints, evictionConstraintsForMinPodAge(args.MinPodAge)...)
-	d.constraints = append(d.constraints, evictionConstraintsForIgnorePodsWithoutPDB(args.IgnorePodsWithoutPDB, handle)...)
 	return nil
 }
 
+// applyEffectivePodProtections applies all relevant protection pod policies to the plugin.
+// For each:
+// - If it's in the list, we disallow eviction (shouldEvict = false).
+// - If it's NOT in the list, we allow eviction (shouldEvict = true).
+func applyEffectivePodProtections(d *DefaultEvictor, podProtections []PodProtection, handle frameworktypes.Handle) error {
+	policyMap := make(map[PodProtection]bool)
+	for _, policy := range podProtections {
+		policyMap[policy] = true
+	}
+
+	d.constraints = append(d.constraints, evictionConstraintsForFailedBarePods(d.logger, !policyMap[FailedBarePods])...)
+	if constraints, err := evictionConstraintsForSystemCriticalPods(d.logger, !policyMap[SystemCriticalPods], d.args.PriorityThreshold, handle); err != nil {
+		return err
+	} else {
+		d.constraints = append(d.constraints, constraints...)
+	}
+	d.constraints = append(d.constraints, evictionConstraintsForLocalStoragePods(!policyMap[PodsWithLocalStorage])...)
+	d.constraints = append(d.constraints, evictionConstraintsForDaemonSetPods(!policyMap[DaemonSetPods])...)
+
+	d.constraints = append(d.constraints, evictionConstraintsForPvcPods(policyMap[PodsWithPVC])...)
+	d.constraints = append(d.constraints, evictionConstraintsForIgnorePodsWithoutPDB(policyMap[PodsWithoutPDB], handle)...)
```

Comment:
Can we try to simplify this function with a `for` and a `switch` statement ?  I think this is confusing to understand.